### DISCUSSION TITLE
hx-on:htmx:before-request="function defined from x-data object"

### DISCUSSION THREAD

**Discussion by ericxin1982** [alpinejs/alpine#3961] [4. General]
Hi there,

Just wondering if that is feasible, the hx-on:htmx:before-request can trigger a function I defined in x-data (alpinejs) above, like:

![image](https://github.com/alpinejs/alpine/assets/9047436/5f740090-cc66-4c24-94ce-02d1f9213a02)

Thanks
Eric Xin

**Comment by ekwoka** [comment]
No.

But you can just do `@htmx:before-request`

**Reply by ericxin1982** [reply]
That is awesome.
